lib/versionmanager.js

```diff
@@ -358,19 +359,17 @@ function queryVersions(packageMap, options = {}) {
 
     // ignore 404 errors from getPackageVersion by having them return null instead of rejecting
     function getPackageVersionProtected(dep) {
-        if (bar) {
-            bar.tick();
-        }
+        bar.tick();
         return getPackageVersion(dep, packageMap[dep], options.pre).catch(err => {
-            if (err && (err.message || err).toString().match(/E404|404 Not Found|doesn't exist/i)) {
+            if (err && (err.message || err).toString().match(/E404|404 Not Found|doesn't exist/i) ||
+                err instanceof packageJson.PackageNotFoundError) {
```

Comment:
On the previous PR I was getting an runtime error trying to access PackageNotFoundError, which was why I removed it, but clearly I was doing something wrong, as this works now. Thank you!

Reply:
I must have somehow had `package-json` v6.0.0 before [errors were exported](https://github.com/sindresorhus/package-json/commit/685fef238e6c5c4e053fda219e96cb42fae0e109).